framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -407,71 +406,47 @@ public S initialStore(UnderlyingAST underlyingAST, @Nullable List<LocalVariableN
+   * #isNotFullyInitializedReceiver(MethodTree)}.
+   *
+   * @param info initial store into which field values are inserted
+   * @param classTree the class that contains {@code methodTree}
+   * @param methodTree the method or constructor tree
+   */
+  private void addFieldValues(S info, ClassTree classTree, MethodTree methodTree) {
+    boolean isConstructor = TreeUtils.isConstructor(methodTree);
+    List<FieldValues<V>> fields = analysis.getFieldValues();
+    TypeElement classEle = TreeUtils.elementFromDeclaration(classTree);
+    for (FieldValues<V> fieldValues : fields) {
+      VariableElement varEle = fieldValues.field.getField();
+      // Insert the value from the initializer of private final fields.
+      if (fieldValues.initializer != null
+          && varEle.getModifiers().contains(Modifier.PRIVATE)
+          && ElementUtils.isFinal(varEle)
+          && analysis.atypeFactory.isImmutable(ElementUtils.getType(varEle))) {
+        info.insertValue(fieldValues.field, fieldValues.initializer);
       }
-    }
 
-    // add properties about fields (static information from type)
-    boolean isNotFullyInitializedReceiver = isNotFullyInitializedReceiver(methodTree);
-    if (isNotFullyInitializedReceiver && !TreeUtils.isConstructor(methodTree)) {
-      // cannot add information about fields if the receiver isn't initialized
-      // and the method isn't a constructor
-      return;
-    }
-    for (Tree member : classTree.getMembers()) {
-      if (member instanceof VariableTree) {
-        VariableTree vt = (VariableTree) member;
-        final VariableElement element = TreeUtils.elementFromDeclaration(vt);
-        AnnotatedTypeMirror type =
-            ((GenericAnnotatedTypeFactory<?, ?, ?, ?>) factory).getAnnotatedTypeLhs(vt);
-        TypeMirror fieldType = ElementUtils.getType(element);
-        JavaExpression receiver;
-        if (ElementUtils.isStatic(element)) {
-          receiver = new ClassName(classType);
-        } else {
-          receiver = new ThisReference(classType);
-        }
-        V value = analysis.createAbstractValue(type);
-        if (value == null) {
-          continue;
-        }
-        if (TreeUtils.isConstructor(methodTree)) {
-          // If we are in a constructor, then we can still use the static type, but only if there is
-          // also an initializer that already does some initialization.
-          boolean found = false;
-          for (Pair<VariableElement, V> fieldValue : fieldValues) {
-            if (fieldValue.first.equals(element)) {
-              value = value.leastUpperBound(fieldValue.second);
-              found = true;
-              break;
-            }
-          }
-          if (!found) {
-            // no initializer found, cannot use static type
-            continue;
-          }
+      // Maybe insert the declared type:
+      // If it's not a constructor, use the declared type if the rreceiver of the method is fully
```

Comment:
Minor: These comments are in a different order ("not a constructor" then "is a constructor") than the comments in the method Javadoc.  It's easier to verify that they match if they are in the same order.  If this one is reordered, the `if` test should probably be restructured so it still matches the comment.